src/Compatibility/Microsoft.DotNet.ApiSymbolExtensions/AssemblySymbolLoader.cs

```diff
@@ -341,11 +357,14 @@ private void ResolveReferences(PEReader peReader, ImmutableHashSet<string>? refe
                 AssemblyReference reference = reader.GetAssemblyReference(handle);
                 string name = $"{reader.GetString(reference.Name)}.dll";
 
+                // Skip assemblies that should never get loaded because they are purely internal
+                if (s_assembliesToIgnore.Contains(name))
```

Comment:
nit: you can save some space and a little bit of work by omitting the `.dll` from all the items in the list and doing this check before appending the file extension.